examples/ai-core/src/lib/present-image.ts

```diff
@@ -0,0 +1,49 @@
+import { Experimental_GeneratedImage as GeneratedImage } from 'ai';
+import fs from 'node:fs';
+import imageType from 'image-type';
+import path from 'node:path';
+import sharp from 'sharp';
+import terminalImage from 'terminal-image';
+
+const OUTPUT_DIR = 'output';
+
+/**
+ * Displays images in the terminal using a downsampled preview and saves the
+ * original, full-resolution files to the output directory with unique
+ * timestamps.
+ * @param images - An array of generated images to process and display.
+ */
+export async function presentImages(images: GeneratedImage[]) {
+  const timestamp = Date.now();
+  for (const [index, image] of images.entries()) {
+    let srcBuffer = image.uint8Array;
+
+    // Determine the format of the image.
+    const format = await imageType(srcBuffer);
```

Comment:
we should have this internally - we have it for mime types.

Reply:
We have `ai/core/util/detect-image-mimetype.ts` but it's not exported. It also doesn't cover as many [file types](https://www.npmjs.com/package/image-type#supported-file-types). We only need `webp` detection currently but I expect we'll use more in time.

Definitely open to using our own and adding more types there over time as we need it. We export a couple of `util` things already:
```
export { cosineSimilarity } from './util/cosine-similarity';
export { simulateReadableStream } from './util/simulate-readable-stream';
```

Let me know if you're ok exporting it and I can send another small PR to swap out the `image-type` module for that instead.